src/main/java/com/andersenlab/lesson2/entity/Ticket.java

```diff
@@ -1,21 +1,34 @@
-package com.andersenlab.lesson2;
+package com.andersenlab.lesson2.entity;
+
+import com.andersenlab.lesson2.abstraction.IdAbler;
+import com.andersenlab.lesson2.abstraction.Printable;
+import com.andersenlab.lesson2.abstraction.Shareable;
+import com.andersenlab.lesson2.abstraction.annotation.NullChecker;
+import com.andersenlab.lesson2.abstraction.annotation.NullableWarning;
+import com.andersenlab.lesson2.util.RandomTicketGenerator;
+import com.andersenlab.lesson2.util.StadiumSector;
 
 import java.math.BigDecimal;
 import java.time.Instant;
 import java.time.LocalDateTime;
 import java.time.ZoneId;
+import java.util.Objects;
 
-public class Ticket {
+public class Ticket extends IdAbler implements Shareable, Printable {
     private final LocalDateTime createdAt = LocalDateTime.now();
+    @NullableWarning
     private String id;
     private String concertHall;
     private String eventCode;
     private long time;
     private boolean isPromo;
+    @NullableWarning
     private StadiumSector stadiumSector;
     private double maxBackpackWeight;
+    @NullableWarning
     private BigDecimal decimalPriceInCents;
 
+    // Constructors
```

Comment:
You should not put comments without the reason, use them to describe some complex logic that is hard to understand, not to tell that constructor is constructor